Antidote/CallsManager.m

```diff
@@ -209,43 +217,64 @@ - (void)controllerDidChangeContent:(RBQFetchedResultsController *)controller
 
 - (void)updateCurrentCallInterface:(OCTCall *)call
 {
+    BOOL isActive = call.status == OCTCallStatusActive;
+    BOOL videoEnabled = call.videoIsEnabled || call.friendSendingVideo;
+
     if (! [self.currentCall.primaryKeyValue isEqualToString:call.uniqueIdentifier]) {
         [self switchViewControllerForCall:call];
     }
 
+    if (isActive && ! [self.currentCallViewController isKindOfClass:[ActiveCallViewController class]]) {
+        [self switchViewControllerForCall:call];
+    }
+
+    if (videoEnabled && isActive && ! [self.currentCallViewController isKindOfClass:[VideoCallViewController class]]) {
+        [self switchViewControllerForCall:call];
+    }
+
+    if (! videoEnabled && isActive && [self.currentCallViewController isKindOfClass:[VideoCallViewController class]]) {
+        [self switchViewControllerForCall:call];
+    }
+
```

Comment:
These three if-s has `isActive` part, it can be moved out.

```objc
if (isActive) {
    BOOL hasWrongClass = ! [self.currentCallViewController isKindOfClass:[ActiveCallViewController class]];
    BOOL shouldEnableVideo = videoEnabled && ! [self.currentCallViewController isKindOfClass:[VideoCallViewController class]];
    BOOL shouldDisableVideo = ! videoEnabled && [self.currentCallViewController isKindOfClass:[VideoCallViewController class]];

    if (hasWrongClass || shouldEnableVideo || shouldDisableVideo) {
        [self switchViewControllerForCall:call];
    }
}
```